Products/CMFPlone/skins/plone_templates/standard_error_message.py

```diff
@@ -27,6 +27,10 @@
 error_traceback = kwargs.get('error_traceback', None)
 error_value = kwargs.get('error_value', None)
 
+if "text/html" not in context.REQUEST.getHeader('Accept', ''):
+    context.REQUEST.RESPONSE.setHeader("Content-Type", "application/json")
+    return '{"error_type": "%s"}' % error_type
+
```

Comment:
I know it would be nice, but that kind of things will be probably handled
by plone.restapi.
What I have done here is a very basic fix, it is in a Python script so we
cannot import many things (for instance I had to serialize the JSON message
manually, I could not do that with an entire traceback), and replacing
standard_error_message.py with a browser view could be very tricky, as it
is called by acquisition everywhere (in OFS.SimpleItem for instance), so I
decided to keep it the way it is, even if we cannot implement everything we
would expect.

On Mon, Feb 29, 2016 at 10:02 AM, Jens W. Klein <notifications@github.com>
wrote:

> In Products/CMFPlone/skins/plone_templates/standard_error_message.py
> <https://github.com/plone/Products.CMFPlone/pull/1417#discussion_r54383331>
> :
>
> > @@ -27,6 +27,10 @@
> >  error_traceback = kwargs.get('error_traceback', None)
> >  error_value = kwargs.get('error_value', None)
> >
> > +if "text/html" not in context.REQUEST.getHeader('Accept', ''):
> > +    context.REQUEST.RESPONSE.setHeader("Content-Type", "application/json")
> > +    return '{"error_type": "%s"}' % error_type
> > +
>
> As a developer I would really love to have the traceback and other error
> values available. But OTOH it make the response large and we want small
> responses in production. Also we would need to take the permissions into
> account.
> What about adding the additional information only if zope is running in
> debug mode (foreground)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/plone/Products.CMFPlone/pull/1417/files#r54383331>.
>